kernel_hardening_checker/engine.py

```diff
@@ -81,6 +81,17 @@ def set_state(self, data: StrOrNone) -> None:
         self.state = data
 
     def check(self) -> None:
+        # handle checks, provided with list()
+        if self.expected.startswith('*') and self.expected.endswith('*'):
+            if  self.state is not None:
+                option = self.expected.strip('*')
+                optlist  = list(self.state.strip('"').split(','))
```

Comment:
What happens if the value is not in quotes.
I would recommend handling this explicitly.

Reply:
the trouble is that i cant make it pretty, much `if`'s and that's no good, how do you think it should looks like?
```Python
        # handle the *special* check
        if self.expected.startswith('*') and self.expected.endswith('*'):
            if self.state is not None:
                if self.state.startswith('"') and self.state.endswith('"'):
                    option = self.expected.strip('*')
                    optlist  = list(self.state.strip('"').split(','))
                    if option in optlist:
                        self.result = f'OK: {self.state}'
                    else:
                        self.result = f'FAIL: "{self.state}"'
                else:
                    self.result = f'FAIL: "{self.state}"'
            return
```

Reply:
No, I would recommend using assertions.

Also, please add much more tests in the engine unit test. We need a 100% code coverage here.